pkg/cluster/mutate_create.go

```diff
@@ -77,6 +86,22 @@ func (m *CreateMutator) Mutate(ctx context.Context, request *v1beta1.AdmissionRe
 		result = append(result, *patch)
 	}
 
+	patch, err = generic.CopyComponentVersionLabelFromAzureClusterCR(ctx, m.ctrlClient, clusterCR.GetObjectMeta(), label.AzureOperatorVersion)
+	if err != nil {
+		return []mutator.PatchOperation{}, microerror.Mask(err)
+	}
+	if patch != nil {
+		result = append(result, *patch)
+	}
+
+	patch, err = generic.EnsureComponentVersionLabelFromRelease(ctx, m.ctrlClient, clusterCR.GetObjectMeta(), "cluster-operator", label.ClusterOperatorVersion)
```

Comment:
This isn't needed. `cluster-operator` isn't able to reconcile `v1alpha3.Cluster` and hence we use `legacy` version of it.

Reply:
Will test it, that would simplify this PR a lot

Reply:
@tuommaki I already see a problem.
After removing the cluster-operator label, the generated `AzureConfig` lacks the same label:

```
apiVersion: provider.giantswarm.io/v1alpha1
kind: AzureConfig
metadata:
   ...
  labels:
    azure-operator.giantswarm.io/version: 5.0.0-alpha4
    cluster.x-k8s.io/cluster-name: drrr6
    giantswarm.io/cluster: drrr6
    giantswarm.io/organization: giantswarm
    release.giantswarm.io/version: 13.0.0-alpha4
```

Reply:
ok nevermind, cluster creation worked like a charm

Reply:
> the generated AzureConfig lacks the same label

That's perfectly fine because in Azure `cluster-operator` is only reconciling `AzureClusterConfig` :slightly_smiling_face: